docs/source/plugin/bot/channels.rst

```diff
@@ -0,0 +1,146 @@
+    everything about a channel when it leaves.
+
+Getting a channel's information
+===============================
+
+To get a channel's information, you need to know its name, with its channel
+prefix (usually ``#``), such as this::
+
+    channel = bot.channels['#channel_name']
+
+With the ``trigger`` object, you can also access the channel object directly
+(assuming the message comes from a channel, which you should check first)::
+
+    channel = bot.channels[trigger.sender]
+
+The ``channel`` object is an instance of :class:`~sopel.tools.target.Channel`,
+which provides the following information:
+
+* its :attr:`~sopel.tools.target.Channel.name`
+* its :attr:`~sopel.tools.target.Channel.topic`
+* its :attr:`~sopel.tools.target.Channel.users`
+* and its users' :attr:`~sopel.tools.target.Channel.privileges`
+
+.. note::
+
+    To check if a message comes from a channel, you have two options:
+
+    1. use the :func:`~sopel.plugin.require_chanmsg` decorator on your plugin
+       callable
+    2. use an ``if`` block in your function to check
+       :attr:`trigger.sender <sopel.trigger.Trigger.sender>`::
+
+           if not trigger.sender.is_nick():
+               # this trigger is from a channel
+
+       See :meth:`Identifier.is_nick() <sopel.tools.Identifier.is_nick>` for
+       more information.
+
+Getting users in a channel
+==========================
+
+To get a list of users in a **channel**, you can use its
+:attr:`~sopel.tools.target.Channel.users` attribute: this is a map of users you
+can iterate over to get all the users::
+
+    for nick, user in channel.users.items():
+        # do something with the nick and the user
+
+You can access one user in a channel with its nick::
+
+    user = channel.users['Nickname']
+
+With the ``trigger`` object, you can also access the user object directly::
+
+    user = channel.users[trigger.nick]
+
+Manage channels
+===============
+
+Like any IRC client, Sopel uses IRC messages to join or leave channels, to kick
+users or to set channel modes. It provides methods you can use to send those
+commands:
+
+* :meth:`bot.join() <sopel.irc.AbstractBot.join>`: to join a channel
+* :meth:`bot.part() <sopel.irc.AbstractBot.part>`: to leave a channel
+* :meth:`bot.kick() <sopel.irc.AbstractBot.kick>`: to kick a user from a
+  channel (this may requires some extra privileges)
+* :meth:`bot.write() <sopel.irc.AbstractBot.write>`: to send any message to IRC
+
+Join & Part
+-----------
+
+If you want the bot to join or leave a channel, the solution is
+straightforward::
+
+    bot.join('#channel_name')
+    bot.part('#channel_name')
+
+For example, you can recreate the ``.join #channel`` command like so::
+
+    @plugin.commands('join')
+    @plugin.require_admin()
+    def my_custom_join(bot, trigger):
+        channel, key = trigger.group(3), trigger.group(4)
+        bot.join(channel, password=key)
+
+In this example, the bot sends the following IRC message to the server:
```

Comment:
Might want to specify the example command (`.join #channel`) here, and leave the above as just "the .join command". Otherwise, the sample input might be too far from the sample output.